internal/fs/buffer/writebuffer.go

```diff
@@ -0,0 +1,18 @@
+package buffer
+
+type WriteBuffer interface {
```

Comment:
This interface should extend the ioReader interface, example [here](https://stackoverflow.com/questions/48643906/how-to-extend-a-go-interface-to-another-interface) .

Reply:
On it. 
I believe you need it for go client methods. Could you please share function signatures for go client changes? Asking so that I can include any such changes which might be missing here. :)

Reply:
as discussed offline, I am not adding the ioReader interface as we may not require it. As @vadlakondaswetha suggested, we can add it later if required while integrating with go storage client interface.